src/components/Authentication/Register.jsx

```diff
@@ -0,0 +1,278 @@
+	const navigate = useNavigate();
+	const [showPassword, setShowPassword] = useState();
+	const [showConfirmPassword, setShowConfirmPassword] = useState();
+	const focusReset = {
+		firstName: false,
+		lastName: false,
+		email: false,
+		password: false,
+		confirmPassword: false,
+	};
+
+	const showPasswordHandler = () =>
+		showPassword ? setShowPassword(false) : setShowPassword(true);
+
+	const showConfirmPasswordHandler = () =>
+		showConfirmPassword
+			? setShowConfirmPassword(false)
+			: setShowConfirmPassword(true);
+
+	return (
+		<form
+			onSubmit={(e) =>
+				registerHandler(e, location, navigate, registerState, authDispatch)
+			}
+			className="input-form register flex-column flex-gap-1 flex-wrap h-auto w-100"
+		>
+			<section
+				className={`input-container flex-column m-5 ${
+					registerState.firstName.length > 0 || registerState.focus.firstName
+						? "focused"
+						: ""
+				}`}
+			>
+				<input
+					id="first-name"
+					className="textbox-content p-5"
+					type="text"
+					name="first-name"
+					onChange={(e) =>
+						setValueHandler(
+							e,
+							"firstName",
+							"UPDATE_FIRST_NAME",
+							registerDispatch
+						)
+					}
+					value={registerState.firstName}
+					onFocus={() =>
+						setFocusHandler(
+							"firstName",
+							true,
+							"UPDATE_FOCUS",
+							registerDispatch,
+							focusReset
+						)
+					}
+					onBlur={() =>
+						setFocusHandler(
+							"firstName",
+							false,
+							"UPDATE_FOCUS",
+							registerDispatch,
+							focusReset
+						)
+					}
+				/>
+				<label htmlFor="first-name" className="textbox-label m-0 px-1">
+					{" "}
+					First Name<span className="required-field">*</span>
+				</label>
+				<sub className="p-2 my-2 inactive-check"></sub>
+			</section>
+
+			<section
+				className={`input-container flex-column m-5 ${
+					registerState.lastName.length > 0 || registerState.focus.lastName
+						? "focused"
+						: ""
+				}`}
+			>
+				<input
+					id="last-name"
+					className="textbox-content p-5"
+					type="text"
+					name="last-name"
+					onChange={(e) =>
+						setValueHandler(e, "lastName", "UPDATE_LAST_NAME", registerDispatch)
+					}
+					value={registerState.lastName}
+					onFocus={() =>
+						setFocusHandler(
+							"lastName",
+							true,
+							"UPDATE_FOCUS",
+							registerDispatch,
+							focusReset
+						)
+					}
+					onBlur={() =>
+						setFocusHandler(
+							"lastName",
+							false,
+							"UPDATE_FOCUS",
+							registerDispatch,
+							focusReset
+						)
+					}
+				/>
+				<label htmlFor="last-name" className="textbox-label m-0 px-1">
+					{" "}
+					Last Name<span className="required-field">*</span>
+				</label>
+				<sub className="p-2 my-2 inactive-check"></sub>
+			</section>
+
+			<section
+				className={`input-container flex-column m-5 ${
+					registerState.email.length > 0 || registerState.focus.email
+						? "focused"
+						: ""
+				}`}
+			>
+				<input
```

Comment:
Not mandatory, just a suggestion- you can create a generic component for input as the code is repeating. This way you will have a reusable component and will have to write less code. 
You can do it later.